.buildkite/pipeline.yml

```diff
@@ -1,3 +1,66 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  SETUP_GVM_VERSION: 'v0.5.0'
+  DOCKER_COMPOSE_VERSION: '1.25.5'
+  ELASTIC_PACKAGE_COMPOSE_DISABLE_ANSI: "true"
+  KIND_VERSION: 'v0.17.0'
+  K8S_VERSION: 'v1.26.0'
+
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - group: "Check and build"
+    key: "tests"
+    steps:
+      - label: ":go: Run check-ci"
+        key: check-ci
+        command: "make check-ci"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+      - label: ":go: Run local build"
+        key: make-local
+        command: "make local"
+        agents:
+          image: "golang:1.20.3"
+          cpu: "8"
+          memory: "4G"
+
+  - group: "Run tests"
+    steps:
+      - label: "Run unit test"
+        key: unit-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go"
+          - "make test-unit"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
+
+
+      - label: "Run integration tests"
+        key: int-test
+        commands:
+          - "chmod +x .buildkite/scripts/setenv.sh"
+          - ".buildkite/scripts/setenv.sh with-go-docker-compose"
+          - "make test-int"
+          - "make junit-report"
+          - "buildkite-agent artifact upload build/*.xml"
+        agents:
+          provider: "gcp"
+
+      - label: ":junit: Junit annotate"
+        commands:
+          - "buildkite-agent artifact download 'build/*.xml' build/"
+        plugins:
+          - junit-annotate#v2.4.1:
+              artifacts: "build/*.xml"
+        agents:
+          provider: "gcp"
+        depends_on:
+          - "unit-test"
+          - "int-test"
+    depends_on: "tests"
```

Comment:
This depends_on forces to wait for the "Run check-ci" and "Run local build" steps ?

Reply:
In my opinion - YES, because if we didn't pass those steps we shouldn't start the next steps.
Of course, we can skip this part, but we can save only up to 1 minute by waiting for these steps (unit-test and int-test).